matlab/test/arrow/tabular/tRecordBatch.m

```diff
@@ -135,5 +117,110 @@ function ErrorIfIndexIsNonPositive(tc)
+
+            arrowArrays = cell([1 width(TOriginal)]);
+            for ii = 1:width(TOriginal)
+                arrowArrays{ii} = arrow.array(TOriginal.(ii));
+            end
+
+            columnNames = string(char(65:77)')';
+            arrowRecordBatch = RecordBatch.fromArrays(arrowArrays{:}, ColumnNames=columnNames);
+            TOriginal.Properties.VariableNames = columnNames;
+            tc.verifyRecordBatch(arrowRecordBatch, columnNames, TOriginal);
+        end
+
+        function FromArraysUnequalArrayLengthsError(tc)
+        % Verify arrow.tabular.RecordBatch.fromArrays throws an error whose
+        % identifier is "arrow:tabular:UnequalArrayLengths" if the arrays
+        % provided don't all have the same length.
+            import arrow.tabular.RecordBatch
+
+            A1 = arrow.array([1, 2]);
+            A2 = arrow.array(["A", "B", "C"]);
+            fcn = @() RecordBatch.fromArrays(A1, A2);
+            tc.verifyError(fcn, "arrow:tabular:UnequalArrayLengths");
+        end
+
+        function FromArraysWrongNumberColumnNamesError(tc)
+        % Verify arrow.tabular.RecordBatch.fromArrays throws an error whose
+        % identifier is "arrow:tabular:WrongNumberColumnNames" if the 
+        % ColumnNames provided doesn't have one element per array.
+            import arrow.tabular.RecordBatch
+
+            A1 = arrow.array([1, 2]);
+            A2 = arrow.array(["A", "B"]);
+            fcn = @() RecordBatch.fromArrays(A1, A2, columnNames=["A", "B", "C"]);
+            tc.verifyError(fcn, "arrow:tabular:WrongNumberColumnNames");
+        end
+
+        function FromArraysColumnNamesHasMissingString(tc)
+        % Verify arrow.tabular.RecordBatch.fromArrays throws an error whose
+        % identifier is "MATLAB:validators:mustBeNonmissing" if the 
+        % ColumnNames provided has a missing string value.
+            import arrow.tabular.RecordBatch
+
+            A1 = arrow.array([1, 2]);
+            A2 = arrow.array(["A", "B"]);
+            fcn = @() RecordBatch.fromArrays(A1, A2, columnNames=["A", missing]);
+            tc.verifyError(fcn, "MATLAB:validators:mustBeNonmissing");
+        end
     end
+
+    methods
+        function verifyRecordBatch(tc, recordBatch, expectedColumnNames, expectedTable)
+            tc.verifyEqual(recordBatch.NumColumns, int32(width(expectedTable)));
+            tc.verifyEqual(recordBatch.ColumnNames, expectedColumnNames);
+            convertedTable = recordBatch.table();
+            tc.verifyEqual(convertedTable, expectedTable);
+             for ii = 1:recordBatch.NumColumns
+                column = recordBatch.column(ii);
+                tc.verifyEqual(column.toMATLAB(), expectedTable{:, ii});
+                traits = arrow.type.traits.traits(string(class(expectedTable{:, ii})));
+                tc.verifyInstanceOf(column, traits.ArrayClassName);
+             end
+        end
+    end
+end
+
+function T = createTableWithAllSupportedTypes()
```

Comment:
Thanks for creating this function. I think it would be good for us to move this into a separate test utility because this table is created in other files too (e.g. [the Feather tests](https://github.com/apache/arrow/blob/dcf913a368b1075111a690ab2c3e7dcb85f758b2/matlab/test/tfeather.m#L223)).

Reply:
Sounds good. Here's the issue #37179.

Reply:
Thanks!